### ISSUE TITLE
grpc-js client fails to process messages due to stream shutdown

### ISSUE THREAD

**Issue by matthewjamesadam** [grpc/grpc-node#2074]
### Problem description
We have a grpc-js client (running against a Kotlin gRPC server https://grpc.io/docs/languages/kotlin/quickstart/).  We're just trying out the simplest possible scenario: the client making a call and seeing a response message.  The problem is that the client gets a status response, but the returned message is always `undefined`.  After tracing through the client behaviour I am seeing the following:

1. The client receives the response message data (`Http2CallStream.push(message)`).  The message is held and processed asynchronously:
```
    push(message) {
        this.trace('pushing to reader message of length ' +
            (message instanceof Buffer ? message.length : null));
        this.canPush = false;
        process.nextTick(() => {
            var _a;
            /* If we have already output the status any later messages should be
             * ignored, and can cause out-of-order operation errors higher up in the
             * stack. Checking as late as possible here to avoid any race conditions.
             */
            if (this.statusOutput) {
                return;
            }
            (_a = this.listener) === null || _a === void 0 ? void 0 : _a.onReceiveMessage(message);
            this.maybeOutputStatus();
        });
    }
```

Before the anonymous function in `nextTick` is called, the following code is run (in `Http2CallStream.attachHttp2Stream`):
```
            stream.on('end', () => {
                this.readsClosed = true;
                this.maybeOutputStatus();
            });
``` 

`maybeOutputStatus` causes the status to be published synchronously, and causes `this.statusOutput` to be set to `true`.  Afterwards, the async anonymous function in `nextTick` above is called, but since the status has already been output, the message is not sent.

This feels like a bug, in that the message is received and processed asynchronously, while stream shutdown is processed synchronously, so the stream shutdown is always processed first and the message is never published.

I am not sure if the stream shutdown is expected or not -- we have not changed or set any particular configuration on either the client or server.  I maybe would expect http keep alive to prevent the stream from shutting down, but I am not sure about this.

### Environment
 - MacOS Monterey
 - Node 14.16.0
 - Node installation method: Running in VSCode electron instance
 - Package name and version: @grpc/grpc-js@1.5.10

**Comment by murgatroid99** [comment]
This does look like a potential issue. Can you run your test client with the environment variables `GRPC_TRACE=all` and `GRPC_VERBOSITY=DEBUG` and share the log output? That will give a clearer order of events in the call handling code.

**Comment by matthewjamesadam** [comment]
Hi @murgatroid99 ,

I actually figured out that the problem was on our end.  Our app builds as a webpack bundle, and we incorrectly had the following in our webpack configuration:

```
    plugins: [
        new webpack.ProvidePlugin({
            process: 'process/browser',
        }),
    ],
```

This effectively replaces the node `process` lib with a polyfill that is meant for browsers.  Something about this polyfill causes the behaviour described above in grpc-js.  Once I removed that polyfill, grpc-js works as expected.
